Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -254,6 +254,25 @@ void CookieStore::get(CookieStoreGetOptions&& options, Ref<DeferredPromise>&& pr
         return;
     }
 
+    if (options.name.isNull() && options.url.isNull()) {
+        promise->reject(ExceptionCode::TypeError);
+        return;
+    }
+
+    if (!options.url.isNull()) {
+        auto parsed = context->completeURL(options.url);
+        if (context->isDocument() && parsed != context->url()) {
+            promise->reject(ExceptionCode::TypeError);
+            return;
+        }
+
+        if (!origin->isSameOriginDomain(SecurityOrigin::create(parsed))) {
```

Comment:
Should use `SecurityOrigin::isSameOriginAs()`

Reply:
I made the change to `SecurityOrigin::isSameOriginAs()` here and in `CookieStore::getAll()` as well.